src/proctored-exam-settings/ProctoredExamSettings.jsx

```diff
@@ -25,11 +25,18 @@ function ExamSettings(props) {
   }
 
   function onCreateZendeskTicketsChange(event) {
-    setAllowOptingOut(event);
+    setCreateZendeskTickets(event);
   }
 
   function onProctoringProviderChange(event) {
-    setProctoringProvider(event.target.value);
+    const provider = event.target.value;
+    setProctoringProvider(provider);
+
+    if (provider === 'proctortrack') {
+      setCreateZendeskTickets(false);
+    } else if (provider === 'software_secure') {
+      setCreateZendeskTickets(true);
+    }
```

Comment:
In reading this it seems very business-logic-y, and I feel myself reaching for the best way to move it to a different file. What do you think?

Reply:
@matthugs I would be supportive of moving this logic to another file/function if there were more too it (or the possibility that more could be added), but the change is small, and as of now this is the only dependency on proctoring provider. Is that along the lines of what you were thinking? Just want to make sure I understand your thinking correctly!